seapath_ros_driver/src/seapath_ros_driver.cpp

```diff
@@ -184,55 +199,67 @@ namespace seapath
         kv6.value = std::to_string(data.heading_error);
 
-        diagnostic_msg.values = {kv1, kv2, kv3, kv4, kv5, kv6};
+        auto flags = data.evaluateDiagnosticStatus();
 
-        if (position_error)
-        {
-            diagnostic_msg.level = diagnostic_msgs::msg::DiagnosticStatus::WARN;
-            diagnostic_msg.message = "Position error";
-        }
-        else if (orientation_error)
+
+        kv7.key = "Horizontal position and velocity status";
+        kv7.value = KMBinaryData::status_to_string(flags.horizontal_pos_vel);
+
+        kv8.key = "Roll and pitch status";
+        kv8.value = KMBinaryData::status_to_string(flags.roll_pitch);
+        
+        kv9.key = "Heading status";
+        kv9.value = KMBinaryData::status_to_string(flags.heading);
+
+        kv10.key = "Heave and vertical velocity status";
+        kv10.value = KMBinaryData::status_to_string(flags.heave_vert_vel);
+
+        kv11.key = "Acceleration status";
+        kv11.value = KMBinaryData::status_to_string(flags.acceleration);
+
+        kv12.key = "Delayed heave status";
+        kv12.value = KMBinaryData::status_to_string(flags.delayed_heave);
+
+        diagnostic_msg.values = {kv1, kv2, kv3, kv4, kv5, kv6, kv7, kv8, kv9, kv10, kv11, kv12};
+
+        // Determine the level and message based on error checks
+        switch (data.determineOverallStatus())
         {
+        case KMBinaryData::DiagnosticStatus::INVALID:
+            diagnostic_msg.level = diagnostic_msgs::msg::DiagnosticStatus::ERROR;
+            diagnostic_msg.message = "Invalid data";
+            break;
+        case KMBinaryData::DiagnosticStatus::REDUCED:
             diagnostic_msg.level = diagnostic_msgs::msg::DiagnosticStatus::WARN;
-            diagnostic_msg.message = "Orientation error";
-        }
-        else
-        {
+            diagnostic_msg.message = "Reduced performance";
+            break;
+        case KMBinaryData::DiagnosticStatus::OK:
             diagnostic_msg.level = diagnostic_msgs::msg::DiagnosticStatus::OK;
-            diagnostic_msg.message = "Socket connection is OK";
+            diagnostic_msg.message = "All systems nominal";
+            break;
         }
-    
+        
         diagnostic_array.status.push_back(diagnostic_msg);
         return diagnostic_array;
     }
 
-    nav_msgs::msg::Odometry Driver::get_odometry_message(const KMBinaryData &data)
+    nav_msgs::msg::Odometry Driver::get_odometry_message(const KMBinaryData& data, const rclcpp::Time& time) const
     {
         nav_msgs::msg::Odometry odom_msg;
-        odom_msg.header.stamp = this->get_clock()->now();
+        odom_msg.header.stamp = time;
 
-        odom_msg.header.frame_id = "world_frame";
-        odom_msg.child_frame_id = "seapath_frame";
-        float north = data.latitude;
-        float east = data.longitude;
-        float height = data.ellipsoid_height;
+        odom_msg.header.frame_id = "seapath";
+        odom_msg.child_frame_id = "seapath";
+       
```

Comment:
make sure this is correct

Reply:
Should be fixed now. Good spot :)

Reply:
Parent is now world frame, and child is seapath. "child_frame_id: This represents the coordinate frame that is moving with the mobile base, typically called the "base frame". This is the frame of reference attached to the robot or vehicle itself". I set it to seapath, could be base_link as well